pydov/util/query.py

```diff
@@ -62,34 +62,43 @@ class Join(PropertyInList):
     'pkey_boring', y), ...]) for every x, y, in df['pkey_boring']
 
     """
-    def __init__(self, dataframe, join_column):
+    def __init__(self, dataframe, on, using=None):
         """Initialisation.
 
         Parameters
         ----------
         dataframe : pandas.DataFrame
             Dataframe to use a basis for joining.
-        join_column : str
-            Name of the column to join on. This column should both exists in
-            the dataframe and in the object type being searched.
+        on : str
+            Name of the column in the queried datatype to join on.
+        using : str, optional
+            Name of the column in the dataframe to use for joining. By
+            default, the same column name as in `on` is assumed.
 
         Raises
         ------
         ValueError
-            If the join_column is not present in the dataframe.
+            If the `using` column is not present in the dataframe.
```

Comment:
specify which dataframe: `not present in the ........ dataframe.`

Reply:
There is only one dataframe: the one that is given in the `dataframe` parameter. The Join is a query builder that builds a query from that dataframe (which is in fact a list of PropertyIsEqualTo inside an Or). It is only when that Join query is used in a .search() of another type that the query (and thus the join) is made.

This also why we cannot validate the `on` column when creating a Join: at this point we don't know yet which datatype will be joined with.

Reply:
ok, that clarifies a lot.